core/server/src/main/java/alluxio/worker/netty/FileDataServerHandler.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.netty;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.network.protocol.RPCFileReadRequest;
+import alluxio.network.protocol.RPCFileReadResponse;
+import alluxio.network.protocol.RPCFileWriteRequest;
+import alluxio.network.protocol.RPCFileWriteResponse;
+import alluxio.network.protocol.RPCResponse;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.network.protocol.databuffer.DataByteBuffer;
+import alluxio.worker.file.FileSystemWorker;
+
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * This class handles filesystem data server requests.
+ */
+@NotThreadSafe
+public class FileDataServerHandler {
```

Comment:
Per our discussion, shall we add a TODO to rename it to "UnderFileDataServerHander"?